### PR TITLE
Plugins: Fix status_source always being "plugin" in plugin request logs

### PR DESCRIPTION
**Pull request by xnyo**
**What is this feature?**

Fixes a bug where `status_source` in plugin request logs is always `plugin`, even if it was set to `downstream` by a plugin.

This happened because the logic for the status source was all contained in the metrics middleware, and the logger middleware is at a deeper level:

```mermaid
flowchart TD
a["MetricsMiddleware pre: Create pluginrequestmeta"]
--> QueryData
--> b["LoggerMiddleware post: Log status_source from context.Context"]
--> c["MetricsMiddleware post: Set status_source in context.Context"]
--> d["MetricsMiddleware post: Instrument using status_source in context.Context"]
```

This PR adds two new middlewares:

- `PluginRequestMetaMiddleware`, which is at the top of the middleware stack, creates the plugin request meta in the `context.Context`, which is then available to all subsequent middlewares
- `StatusSourceMiddleware`, at the bottom of the middleware stack, which sets the status source in the `context.Context` before returning, so all middlewares above it can access the correct status source *after* QueryData has returned

```mermaid
flowchart TD
a["PluginRequestMetaMiddleware: Create pluginrequestmeta"]
--> QueryData
--> c["StatusSourceMiddleware post: Set status_source in context.Context"]
--> b["LoggerMiddleware post: Log status_source from context.Context"]
--> d["MetricsMiddleware post: Instrument using status_source in context.Context"]
```

**Why do we need this feature?**

Bugfix

**Who is this feature for?**

Grafana operators

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by xnyo** [comment]
cc @ivanahuckova

**Comment by ivanahuckova** [comment]
Thank you @xnyo for looking into this and fixing it! 🧡

File: pkg/services/pluginsintegration/clientmiddleware/status_source_middleware.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+		}
+	}
+
+	// A plugin error has higher priority than a downstream error,
+	// so set to downstream only if there's no plugin error
+	if hasDownstreamError && !hasPluginError {
+		if err := pluginrequestmeta.WithDownstreamStatusSource(ctx); err != nil {
+			return resp, fmt.Errorf("failed to set downstream status source: %w", err)
+		}
+	}
```
Comment (xnyo) [inline pkg/services/pluginsintegration/clientmiddleware/status_source_middleware.go:56]: This is the same logic as before, but it has been moved from MetricsMiddleware

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**